### PR TITLE
update examples to use kebab case

### PR DESCRIPTION
**Pull request by dprotaso**
This makes the config maps and annotation keys consistent 

- update serving examples to use kebab-case for annotations
- update autoscaling examples to use kebab-case for annotations
- update examples to use kebab case networking annotations
- update operator config example to use kebab-case

Related:
- https://github.com/knative/serving/pull/12329
- https://github.com/knative/serving/issues/12103

### CODE REVIEW

**Review by psschwei — COMMENTED** [review]
I found a couple of other places that might need to also be updated:

* https://github.com/knative/docs/blob/main/docs/serving/config-ha.md#scaling-the-data-plane
* https://github.com/knative/docs/blob/main/docs/serving/load-balancing/activator-capacity.md#configuring-activator-capacity

File: docs/serving/using-auto-tls.md
Comment (dprotaso) [reply docs/serving/using-auto-tls.md:165]: good catch

**Comment by dprotaso** [comment]
> https://github.com/knative/docs/blob/main/docs/serving/config-ha.md#scaling-the-data-plane

This looks ok - since the minScale, maxScale is being applied to the K8s HPA for the activator

**Review by psschwei — APPROVED** [review]
/lgtm
I think this is probably ok without a review from the docs team (since the changes are config not copy), but just to be safe
/assign @snneji 
/hold

**Comment by snneji** [comment]
looks fine to me. 
/unhold

**Pull request merged**
